core/src/main/java/com/datastrato/gravitino/storage/relational/mysql/MySQLBackend.java

```diff
@@ -25,41 +38,207 @@ public class MySQLBackend implements RelationalBackend {
 
   /** Initialize the MySQL backend instance. */
   @Override
-  public void initialize(Config config) {}
+  public void initialize(Config config) {
+    SqlSessionFactoryHelper.getInstance().init(config);
+  }
 
   @Override
   public <E extends Entity & HasIdentifier> List<E> list(
-      Namespace namespace, Entity.EntityType entityType) throws NoSuchEntityException {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+      Namespace namespace, Entity.EntityType entityType) {
+    try (SqlSession session = SqlSessions.getSqlSession()) {
+      try {
+        switch (entityType) {
+          case METALAKE:
+            List<MetalakePO> metalakePOS =
+                ((MetalakeMetaMapper) SqlSessions.getMapper(MetalakeMetaMapper.class))
+                    .listMetalakePOs();
+            return metalakePOS != null
+                ? metalakePOS.stream()
+                    .map(
+                        metalakePO -> {
+                          try {
+                            return (E) POConverters.fromMetalakePO(metalakePO);
+                          } catch (JsonProcessingException e) {
+                            throw new RuntimeException(e);
+                          }
+                        })
+                    .collect(Collectors.toList())
+                : new ArrayList<>();
+          case CATALOG:
+          case SCHEMA:
+          case TABLE:
+          case FILESET:
+          default:
+            throw new IllegalArgumentException(
+                String.format("Unsupported entity type: %s for list operation", entityType));
+        }
+      } catch (Throwable t) {
+        throw new RuntimeException(t);
+      } finally {
+        SqlSessions.closeSqlSession();
+      }
+    }
   }
 
   @Override
   public boolean exists(NameIdentifier ident, Entity.EntityType entityType) {
-    throw new UnsupportedOperationException("Unsupported operation now.");
+    try (SqlSession session = SqlSessions.getSqlSession()) {
```

Comment:
Could you extract a method for this logic? This logic repeated multiple times. You can refer to the method `PrincipalUtils#doAs`.

Reply:
The relevant logic will be extracted as a tool class